File: mobile/buildflow_mobile/lib/services/inventory_service.dart
import 'api_service.dart';

class InventoryService {
  InventoryService(this.api);
  final ApiService api;

  Future<List<Map<String, dynamic>>> materials({String? search}) async {
    final query = search == null || search.isEmpty ? '' : '&search=${Uri.encodeQueryComponent(search)}';
    final data = await api.request('GET', '/inventory/materials/page?page=1&pageSize=100$query') as Map<String, dynamic>;
    return (data['items'] as List).cast<Map<String, dynamic>>();
  }

  Future<List<Map<String, dynamic>>> alerts() async =>
      (await api.request('GET', '/inventory/alerts/low-stock?threshold=0') as List).cast<Map<String, dynamic>>();

  Future<Map<String, dynamic>> receive(String materialId, double quantity, {String? reference}) async =>
      (await api.request('POST', '/inventory/materials/$materialId/receive', body: {
        'quantity': quantity,
        'reference': reference ?? 'site receipt',
      })) as Map<String, dynamic>;

  Future<Map<String, dynamic>> checkAvailability(List<Map<String, dynamic>> requirements) async =>
      (await api.request('POST', '/inventory/reservations/check', body: requirements)) as Map<String, dynamic>;
}

File: frontend/buildflow-web/src/features/inventory/inventoryService.js
import { api } from '../../services/api.js'

export const inventoryService = {
  materials: async (params = {}) => (await api.get('/inventory/materials/page', { params })).data,
  material: async (id) => (await api.get(`/inventory/materials/${id}`)).data,
  createMaterial: async (body) => (await api.post('/inventory/materials', body)).data,
  updateMaterial: async (id, body) => (await api.put(`/inventory/materials/${id}`, body)).data,
  deleteMaterial: async (id) => api.delete(`/inventory/materials/${id}`),
  warehouses: async (params = {}) => (await api.get('/inventory/warehouses/page', { params })).data,
  createWarehouse: async (body) => (await api.post('/inventory/warehouses', body)).data,
  updateWarehouse: async (id, body) => (await api.put(`/inventory/warehouses/${id}`, body)).data,
  deleteWarehouse: async (id) => api.delete(`/inventory/warehouses/${id}`),
  stock: async (id, action, body) => (await api.post(`/inventory/materials/${id}/${action}`, body)).data,
  alerts: async (threshold = 0) => (await api.get('/inventory/alerts/low-stock', { params: { threshold } })).data,
  reservations: async (params = {}) => (await api.get('/inventory/reservations', { params })).data,
  reserve: async (id, body) => (await api.post(`/inventory/materials/${id}/reservations`, body)).data,
  release: async (id) => api.delete(`/inventory/reservations/${id}`),
  movements: async (params = {}) => (await api.get('/inventory/movements', { params })).data,
}